### DISCUSSION TITLE
bunx/npx/pnpm dlx prisma db push won't finish setup

### DISCUSSION THREAD

**Discussion by caiohportella** [prisma/prisma#23905] [Q&A]
### Question

Every time I run the command `prisma db push` or `prisma migrate dev`, the terminal won't progress, it stales at the same stage and my model never gets pushed, like so:

![terminal](https://github.com/prisma/prisma/assets/63035330/d185bcc1-c6ed-4715-8044-88d2f5296366)

This occurs independent if I'm using npx, pnpm dlx or bunx. That request is never fulfilled and I never get a database to work with. Could anyone help me solve this?

### How to reproduce (optional)

1. Clone the repo https://github.com/caiohportella/quill
2. Run bun install
3. Run `bunx prisma db push`
4. The terminal won't load, as like my screenshot

### Expected behavior (optional)

Push my model to the database.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model User {
  id    String @id @unique
  email String @unique

  stripeCustomerId       String?   @unique @map("stripe_customer_id")
  stripeSubscriptionId   String?   @unique @map("stripe_subscription_id")
  stripePriceId          String?   @map("stripe_price_id")
  stripeCurrentPeriodEnd DateTime? @map("stripe_current_period_end")
}

```

```ts
import { PrismaClient } from '@prisma/client'

declare global {
  var cachedPrisma: PrismaClient
}

let prisma: PrismaClient
if (process.env.NODE_ENV === 'production') {
  prisma = new PrismaClient()
} else {
  if (!global.cachedPrisma) {
    global.cachedPrisma = new PrismaClient()
  }
  prisma = global.cachedPrisma
}

export const db = prisma
```

- OS: Windows 10
- Database: PostgreSQL
- Node.js version: v20.9.0
- prisma                  : 5.12.1
- @prisma/client          : 5.12.1
- Computed binaryTarget   : windows
- Operating System        : win32
- Architecture            : x64
- Node.js                 : v20.9.0
- Query Engine (Node-API) : libquery-engine 473ed3124229e22d881cb7addf559799debae1ab (at node_modules\@prisma\engines\query_engine-windows.dll.node)
- Schema Engine           : schema-engine-cli 473ed3124229e22d881cb7addf559799debae1ab (at node_modules\@prisma\engines\schema-engine-windows.exe)
- Schema Wasm             : @prisma/prisma-schema-wasm 5.12.0-21.473ed3124229e22d881cb7addf559799debae1ab
- Default Engines Hash    : 473ed3124229e22d881cb7addf559799debae1ab
- Studio                  : 0.499.0

**Comment by ludralph** [comment]
HI @caiohportella 👋 

Thank you for raising this question.

I was not able to reproduce your issue as the command `bunx prisma db push` successfully pushed my model to the database as shown in the screenshot below.

<img width="1091" alt="image" src="https://github.com/prisma/prisma/assets/13855608/ce7c002e-2877-4095-a123-067dea82eb8d">

You may need to check your network if you are behind a firewall or VPN?

**Reply by caiohportella** [reply]
That's really strange! I'm using my internet normally and also working on other projects with bun where I haven't had this issue. Also, I'm not using any kind of VPN. Any suggestion on how I can get this fixed?

**Reply by ludralph** [reply]
Can you enable [debugging](https://www.prisma.io/docs/orm/prisma-client/debugging-and-troubleshooting/debugging) to see if we can get more information from there?

**Reply by caiohportella** [reply]
Yeah, I tried running `set DEBUG="prisma*"` on console, but when looking through `bunx prisma debug` the variable didn't change. Am I doing this wrong?

**Reply by caiohportella** [reply]
I've done this, take a look:
![terminal](https://github.com/prisma/prisma/assets/63035330/dd0d7fb6-c33c-4786-9365-7372192932b5)

**Reply by ludralph** [reply]
Hi @caiohportella 

Can you check that the versions of `prisma` and `@prisma/client` in your package.json file match. Mismatched versions can sometimes cause issues. As suggested in this [GitHub comment](https://github.com/prisma/prisma/issues/6926), try removing your node_modules folder, then run npm install (or bun install in your case) again. Try running the command with the DEBUG environment variable set to see more detailed output. For example:
```
DEBUG="*" bunx prisma db push
```
This might give you more insight into where the process is hanging.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
